src/pip/_internal/vcs/__init__.py

```diff
@@ -213,10 +213,21 @@ def export(self, location):
         """
         raise NotImplementedError
 
+    def parse_netloc(self, netloc):
+        """
+        Parse the repository URL's netloc, and return the new netloc to use
```

Comment:
I guess it should be mentioned here why some VCS backends don't extract username, password while others might.

Reply:
Good suggestion, thanks. PR updated.